src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.groovy

```diff
@@ -225,6 +229,23 @@ class Application {
         }
     }
 
+    private void determineOperatingSystem() {
```

Comment:
Could it be worthwhile for us to have a component that performs this work and could be polled for, rather than having Application do this and pass it into various inits?